modules/apps/27-interchain-accounts/host/types/params.go

```diff
@@ -46,29 +32,7 @@ func (p Params) Validate() error {
 	return validateAllowlist(p.AllowMessages)
 }
 
-// ParamSetPairs implements params.ParamSet
-func (p *Params) ParamSetPairs() paramtypes.ParamSetPairs {
-	return paramtypes.ParamSetPairs{
-		paramtypes.NewParamSetPair(KeyHostEnabled, p.HostEnabled, validateEnabledType),
-		paramtypes.NewParamSetPair(KeyAllowMessages, p.AllowMessages, validateAllowlist),
-	}
-}
-
-func validateEnabledType(i interface{}) error {
-	_, ok := i.(bool)
-	if !ok {
-		return fmt.Errorf("invalid parameter type: %T", i)
-	}
-
-	return nil
-}
-
-func validateAllowlist(i interface{}) error {
-	allowMsgs, ok := i.([]string)
-	if !ok {
-		return fmt.Errorf("invalid parameter type: %T", i)
-	}
-
+func validateAllowlist(allowMsgs []string) error {
```

Comment:
the code:
```go
    if err := validateEnabledType(p.HostEnabled); err != nil {
	return err
    }
```

can be removed in my opinion. I believe this code only executed as extra defense to how the x/params module worked. Given self managed params, there should be no instance when the params type has a non boolean value for the enabled type (I'm not sure how this could occur)

This would also allow `params_legacy.go` to be deleted without additional changes as the code is written currently

Reply:
Thanks, I was confused why this was there, I'll push this to all other param migration PRs as well